account_partner_journal/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def action_view_invoice(self, invoices=False):
+        """
+        Override method to set default invoice journal as supplier
+        journal
+        """
+        result = super().action_view_invoice(invoices)
+
+        # Use the default Purchase Journal set in Partner for Invoice Journal
+        default_journal = (
```

Comment:
```suggestion
        default_journal = self.partner_id.default_purchase_journal_id
```
Just simple like this